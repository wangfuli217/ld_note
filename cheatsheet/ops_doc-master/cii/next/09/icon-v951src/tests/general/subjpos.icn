#SRC: JCON

procedure main()
   local i, s, x

   ws()		# dump initial state
   write()

   every setsubj(s := "cat" | "" | "donut" | "x") do
      every i := -*s-2 to *s+3 do
         setpos(i)
   write()

   "outer" ? {
      ws()
      &pos := 3
      ws()
      "inner" ? {
         ws()
         &pos := 2
         ws()
         }
      ws()
      }

   "123456" ? {
      &pos := 3
      x := 4
      write("&pos=", &pos, " x=", x)
      &pos :=: x
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 9
      &pos :=: x
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 9
      x :=: &pos
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 9
      every &pos <-> x do write("&pos=", &pos, " x=", x)
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 9
      every x <-> &pos do write("&pos=", &pos, " x=", x)
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 2
      every &pos <-> x do &subject := "A"
      write("&pos=", &pos, " x=", x)
      &pos := 3
      x := 2
      every x <-> &pos do &subject := "A"
      write("&pos=", &pos, " x=", x)
   }
end

procedure setpos(n)
   writes("p := ", right(image(n),10))
   if &pos := n then
      writes("       ")
   else
      writes(" FAILED")
   ws()
   return
end

procedure setsubj(s)
   writes("s := ", right(image(s),10))
   if &subject := s then
      writes("       ")
   else
      writes(" FAILED")
   ws()
   return
end

procedure ws()	# write state of &setpos and &subject
   write("  &pos=", image(&pos), "   &subject=", image(&subject))
   return
end
