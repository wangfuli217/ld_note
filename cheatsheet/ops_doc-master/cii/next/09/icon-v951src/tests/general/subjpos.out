  &pos=1   &subject=""

s :=      "cat"         &pos=1   &subject="cat"
p :=         -5 FAILED  &pos=1   &subject="cat"
p :=         -4 FAILED  &pos=1   &subject="cat"
p :=         -3         &pos=1   &subject="cat"
p :=         -2         &pos=2   &subject="cat"
p :=         -1         &pos=3   &subject="cat"
p :=          0         &pos=4   &subject="cat"
p :=          1         &pos=1   &subject="cat"
p :=          2         &pos=2   &subject="cat"
p :=          3         &pos=3   &subject="cat"
p :=          4         &pos=4   &subject="cat"
p :=          5 FAILED  &pos=4   &subject="cat"
p :=          6 FAILED  &pos=4   &subject="cat"
s :=         ""         &pos=1   &subject=""
p :=         -2 FAILED  &pos=1   &subject=""
p :=         -1 FAILED  &pos=1   &subject=""
p :=          0         &pos=1   &subject=""
p :=          1         &pos=1   &subject=""
p :=          2 FAILED  &pos=1   &subject=""
p :=          3 FAILED  &pos=1   &subject=""
s :=    "donut"         &pos=1   &subject="donut"
p :=         -7 FAILED  &pos=1   &subject="donut"
p :=         -6 FAILED  &pos=1   &subject="donut"
p :=         -5         &pos=1   &subject="donut"
p :=         -4         &pos=2   &subject="donut"
p :=         -3         &pos=3   &subject="donut"
p :=         -2         &pos=4   &subject="donut"
p :=         -1         &pos=5   &subject="donut"
p :=          0         &pos=6   &subject="donut"
p :=          1         &pos=1   &subject="donut"
p :=          2         &pos=2   &subject="donut"
p :=          3         &pos=3   &subject="donut"
p :=          4         &pos=4   &subject="donut"
p :=          5         &pos=5   &subject="donut"
p :=          6         &pos=6   &subject="donut"
p :=          7 FAILED  &pos=6   &subject="donut"
p :=          8 FAILED  &pos=6   &subject="donut"
s :=        "x"         &pos=1   &subject="x"
p :=         -3 FAILED  &pos=1   &subject="x"
p :=         -2 FAILED  &pos=1   &subject="x"
p :=         -1         &pos=1   &subject="x"
p :=          0         &pos=2   &subject="x"
p :=          1         &pos=1   &subject="x"
p :=          2         &pos=2   &subject="x"
p :=          3 FAILED  &pos=2   &subject="x"
p :=          4 FAILED  &pos=2   &subject="x"

  &pos=1   &subject="outer"
  &pos=3   &subject="outer"
  &pos=1   &subject="inner"
  &pos=2   &subject="inner"
  &pos=3   &subject="outer"
&pos=3 x=4
&pos=4 x=3
&pos=3 x=9
&pos=3 x=3
&pos=3 x=9
&pos=3 x=3
&pos=1 x=3
&pos=1 x=2
